### ISSUE TITLE
Where does the upload go?

### ISSUE THREAD

**Issue by os6sense** [actions/upload-artifact#2]
Where does the upload go to? Is there any meta information returned on success/failure? Can we control where the artifact is uploaded? 

Sorry silly questions but the readme is a little on the slim side :)

**Comment by lannonbr** [comment]
In the top right corner of a workflow run, once the run is over, if you used this action, there will be a `Artifacts` dropdown which you can download items from. Here's a screenshot of what it looks like

![Upload example](https://user-images.githubusercontent.com/3685876/62906968-1b4aff80-bd3f-11e9-8815-9058eb05692a.png)

**Comment by TingluoHuang** [comment]
it's uploaded to GitHub server as @lannonbr point out in the screenshot

**Comment by mastoj** [comment]
This is so cool, but I do have a related question. How do you access the artifact programmatically? I can't see it as part of the `check-run`. I have another action where I would like to access the artifact from a specific run based on the git sha. The reason be is that we are using chat ops, so we want to build it on check-in, and then access the artifact when we say that we want to deploy a specific branch.

**Comment by mastoj** [comment]
@TingluoHuang maybe you can answer ⬆️ ?

**Comment by mastoj** [comment]
What I've done so far is GET on the `check-suites` but that doesn't give me any information on where the artifacts are.

**Comment by os6sense** [comment]
@mastoj I'm assuming we use the associated https://github.com/actions/download-artifact project but it would be nice to have more detail :)

**Comment by mastoj** [comment]
@os6sense, haven't seen that. Does that works between different workflows? Have to try I guess.

**Comment by mastoj** [comment]
Seems like it doesn't work to download between workflows :(, so back at square 1.

**Comment by TingluoHuang** [comment]
@mastoj we don't support download artifact from a different check-run yet, it's on our backlog.

**Comment by warrenbuckley** [comment]
> In the top right corner of a workflow run, once the run is over, if you used this action, there will be a `Artifacts` dropdown which you can download items from. Here's a screenshot of what it looks like
> 
> ![Upload example](https://user-images.githubusercontent.com/3685876/62906968-1b4aff80-bd3f-11e9-8815-9058eb05692a.png)

Can I suggest the readme.md is updated with this info as its a bit sparse, as I had no clue what this done or what or where the file was going to upload to please :)

**Comment by TingluoHuang** [comment]
@warrenbuckley we accept PR. 😄

**Comment by warrenbuckley** [comment]
@TingluoHuang here you go - https://github.com/actions/upload-artifact/pull/4 😄

**Comment by TingluoHuang** [comment]
thanks @warrenbuckley

**Comment by gschier** [comment]
I was confused by this as well. Looks like it only shows on this view for some reason: https://github.com/Girchi/girchi-com-frontend/pull/2/checks

I'm uploading on a Release trigger (not a PR) so I'm not able to see the uploads as far as I can tell. It would be AWESOME if the uploads when directly onto the release that triggered the action.

**Comment by bokuweb** [comment]
Is there a way to get Artifact URL from a program? Or are you planning to publish the API?

**Comment by teohhanhui** [comment]
Unfortunately, I can't seem to find the "Artifacts" dropdown, no matter which view I access the run from.

**Comment by joshmgross** [comment]
👋 Hi all, sorry for the lack of response to this issue.

>Where does the upload go to?

As answered above, artifacts will appear in the top-right when looking at a workflow run. Note that a run must be completed before artifacts show up.


>How do you access the artifact programmatically?

>Is there a way to get Artifact URL from a program? Or are you planning to publish the API?

You can access artifacts via the Actions API, see https://developer.github.com/v3/actions/artifacts/
Just note that the download URL will be short-lived (~a few minutes at most) and allow anonymous access for that time period.

Please let me know if there are any more questions.

**Comment by ZacSweers** [comment]
@joshmgross would it be possible to make that url available to subsequent actions somehow?

**Comment by teohhanhui** [comment]
@ZacSweers That's https://github.com/actions/download-artifact/issues/3

**Comment by ZacSweers** [comment]
@teohhanhui I don't see any actual answers in that thread. It seems like starting with having some sort of accessible reference to the previously uploaded artifact on this action would be a good start.

**Comment by joshmgross** [comment]
@ZacSweers I believe @teohhanhui meant that https://github.com/actions/download-artifact/issues/3 is tracking that feature.

Going to close out this issue, please open issues if you have any further questions (this will make it easier for others to find those questions and answers)
